.eslintrc.js

```diff
@@ -6,33 +6,76 @@ module.exports = {
+                'no-restricted-imports': [
+                    'error',
                     {
-                        name: 'react-native',
-                        importNames: ['useWindowDimensions'],
-                        message: 'Please use useWindowDimensions from src/hooks/useWindowDimensions instead',
+                        paths: [
+                            {
+                                name: 'react-native',
+                                importNames: ['useWindowDimensions'],
+                                message: 'Please use useWindowDimensions from src/hooks/useWindowDimensions instead.',
+                            },
+                            {
+                                name: 'react-native',
+                                importNames: ['TouchableOpacity', 'TouchableWithoutFeedback', 'TouchableNativeFeedback', 'TouchableHighlight'],
+                                message: 'Please use PressableWithFeedback and/or PressableWithoutFeedback from src/components/Pressable instead.',
+                            },
+                        ],
                     },
+                ],
+            },
+        },
+        {
+            files: ['*.ts', '*.tsx'],
+            extends: ['expensify', 'airbnb-typescript', 'plugin:storybook/recommended', 'plugin:react-hooks/recommended', 'plugin:@typescript-eslint/recommended', 'prettier'],
+            plugins: ['@typescript-eslint', 'react-hooks'],
+            parser: '@typescript-eslint/parser',
+            parserOptions: {
+                project: './tsconfig.json',
+            },
+            rules: {
+                'import/no-extraneous-dependencies': 'off',
+                'rulesdir/onyx-props-must-have-default': 'off',
+                '@typescript-eslint/no-var-requires': 'off',
+                '@typescript-eslint/no-non-null-assertion': 'error',
+                '@typescript-eslint/no-explicit-any': 'error',
+                '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+                'es/no-nullish-coalescing-operators': 'off',
+                'es/no-optional-chaining': 'off',
+                'no-restricted-syntax': [
+                    'error',
                     {
-                        name: 'react-native',
-                        importNames: ['TouchableOpacity', 'TouchableWithoutFeedback', 'TouchableNativeFeedback', 'TouchableHighlight'],
-                        message: 'Please use PressableWithFeedback and/or PressableWithoutFeedback from src/components/Pressable instead',
+                        selector: 'TSEnumDeclaration',
+                        message: "Please don't declare enums, use union types instead.",
+                    },
+                ],
+                'no-restricted-imports': [
+                    'error',
+                    {
+                        paths: [
+                            {
+                                name: 'lodash/get',
```

Comment:
It does not prevent this case. We could include this rule:
```
{
    name: 'lodash',
    importNames: ['get'],
    message: 'Please use optional chaining and nullish coalescing instead.',
},
```

However the downside is that `import * as _ from 'lodash'` won't work as eslint errors with: `* import is invalid because 'get' from 'lodash' is restricted.`

Reply:
ah I asee

@blazejkustra we also (somehow) use underscore too so we should prevent the direct import of `get` from underscore

Reply:
Oh that's funny 😆 

Turns out that we don't import lodash like `import * as _ from 'lodash'` so we can include these rules:

```
{
    name: 'lodash',
    importNames: ['get'],
    message: 'Please use optional chaining and nullish coalescing instead.',
},
{
    name: 'underscore',
    importNames: ['get'],
    message: 'Please use optional chaining and nullish coalescing instead.',
},
```

I added new rules to eslint to restrict these imports, both lodash and underscore.